homeassistant/components/onvif/switch.py

```diff
@@ -0,0 +1,108 @@
+"""ONVIF switches for controlling cameras."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .base import ONVIFBaseEntity
+from .const import DOMAIN
+from .device import ONVIFDevice
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a ONVIF switch platform."""
+    device = hass.data[DOMAIN][config_entry.unique_id]
+
+    entities = [
+        ONVIFWiperSwitch(device),
+        ONVIFAutoFocusSwitch(device),
+        ONVIFInfraredSwitch(device),
+    ]
+    async_add_entities(entities)
+
+
+class ONVIFImagingSettingSwitch(ONVIFBaseEntity, SwitchEntity):
+    """An ONVIF switch controlled via ImagingSettings."""
+
+    _on_settings: dict[str, Any] = {}
+    _off_settings: dict[str, Any] = {}
+
+    async def async_turn_on(self, **kwargs) -> None:
+        """Turn on switch."""
+        self._attr_is_on = True
+        profile = self.device.profiles[0]
+        await self.device.async_set_imaging_settings(profile, self._on_settings)
+
+    async def async_turn_off(self, **kwargs) -> None:
+        """Turn off switch."""
+        self._attr_is_on = False
+        profile = self.device.profiles[0]
+        await self.device.async_set_imaging_settings(profile, self._off_settings)
+
+
+class ONVIFAuxSwitch(ONVIFBaseEntity, SwitchEntity):
```

Comment:
I suggest using entity descriptions and a single switch entity class instead of multiple classes. Pass the `device` and `profile` parameters to the turn on / off lambdas of the description.

Example:

https://github.com/home-assistant/core/blob/cbcff6435f4efdfd5e92aa5394f776b990666726/homeassistant/components/adguard/switch.py#L24-L48

https://github.com/home-assistant/core/blob/cbcff6435f4efdfd5e92aa5394f776b990666726/homeassistant/components/adguard/switch.py#L113-L145

Reply:
Good idea. Ok I think I have accomplished this in the latest commit.